### DISCUSSION TITLE
Possibility to disable emitting events from FilesService?

### DISCUSSION THREAD

**Discussion by keesvanbemmel** [directus/directus#10187] [Archive]
Hi all,

I'm creating files using the internal FilesService but I want to be able to disable events so that hooks aren't fired.

I think, but I'm not sure, I should handle that [HERE](https://github.com/directus/directus/blob/3d0e086c5501487ab62e31bbf240c20f718d95a9/api/src/services/files.ts#L127)

Is that correct? If so I'm more than happy to create a PR for this!

**Comment by rijkvanzanten** [comment]
That's correct! Some services already have an options parameter in the method that allows skipping the events, like this:

https://github.com/directus/directus/blob/3d0e086c5501487ab62e31bbf240c20f718d95a9/api/src/services/items.ts#L205-L207

If we add the same to the files service, you could disable the event by passing `emitEvents: false` to the method
